### ISSUE TITLE
MSBuild property evaluation failed

### ISSUE THREAD

**Issue by mrward** [mono/monodevelop#7023]
```
ERROR [2019-02-01 11:04:08Z]: MSBuild property evaluation failed: [System.String]::Copy('%(Identity)').Substring(
          0,
          $([System.String]::Copy('%(Identity)').LastIndexOf(
            '.designer.cs',
            System.StringComparison.OrdinalIgnoreCase))
        )
System.FormatException: Input string was not in a correct format.
  at System.Number.StringToNumber (System.ReadOnlySpan`1[T] str, System.Globalization.NumberStyles options, System.Number+NumberBuffer& number, System.Globalization.NumberFormatInfo info, System.Boolean parseDecimal) [0x00053] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-08/external/bockbuild/builds/mono-x64/external/corefx/src/Common/src/CoreLib/System/Number.Parsing.cs:872 
  at System.Number.ParseInt32 (System.ReadOnlySpan`1[T] s, System.Globalization.NumberStyles style, System.Globalization.NumberFormatInfo info) [0x0000a] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-08/external/bockbuild/builds/mono-x64/external/corefx/src/Common/src/CoreLib/System/Number.Parsing.cs:308 
  at System.Int32.Parse (System.String s, System.Globalization.NumberStyles style, System.IFormatProvider provider) [0x00010] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-08/external/bockbuild/builds/mono-x64/external/corefx/src/Common/src/CoreLib/System/Int32.cs:137 
  at System.Convert.ToInt32 (System.String value, System.IFormatProvider provider) [0x00005] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-08/external/bockbuild/builds/mono-x64/external/corefx/src/Common/src/CoreLib/System/Convert.cs:1221 
  at System.String.System.IConvertible.ToInt32 (System.IFormatProvider provider) [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-08/external/bockbuild/builds/mono-x64/external/corefx/src/Common/src/CoreLib/System/String.cs:690 
  at System.Convert.ChangeType (System.Object value, System.Type conversionType, System.IFormatProvider provider) [0x000e0] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-08/external/bockbuild/builds/mono-x64/external/corefx/src/Common/src/CoreLib/System/Convert.cs:356 
  at MonoDevelop.Projects.MSBuild.MSBuildEvaluationContext.ConvertArg (System.Reflection.MethodBase method, System.Int32 argNum, System.Object value, System.Type parameterType) [0x000b3] in /Users/vsts/agent/2.144.2/work/1/s/monodevelop/main/src/core/MonoDevelop.Core/MonoDevelop.Projects.MSBuild/MSBuildEvaluationContext.cs:866 
  at MonoDevelop.Projects.MSBuild.MSBuildEvaluationContext.EvaluateMethod (System.ReadOnlySpan`1[T] str, System.Reflection.MemberInfo[] member, System.Object instance, System.Object[] parameterValues, System.Object& val) [0x00104] in /Users/vsts/agent/2.144.2/work/1/s/monodevelop/main/src/core/MonoDevelop.Core/MonoDevelop.Projects.MSBuild/MSBuildEvaluationContext.cs:719 
```

> VS bug [#783960](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/783960)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-13] Comment by **xamarinc**



Fixed in version 8.4.0.602 (master)



Pull Request #8274 merged by: Lluis Sanchez  
 Author: mono  
 Commit: 4e1a698f255f9cc3997ae9f992abc5777d12c505 (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-15] Comment by **xamarinc**



Fixed in version 8.3.0.1047 (release-8.3)



Pull Request #8274 merged by: Lluis Sanchez  
 Author: mono  
 Commit: 4e1a698f255f9cc3997ae9f992abc5777d12c505 (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-17] Comment by **xamarinc**



Fixed in version 8.6.0.802 (master)



Pull Request #8274 merged by: Lluis Sanchez  
 Author: mono  
 Commit: 4e1a698f255f9cc3997ae9f992abc5777d12c505 (xamarin/monodevelop)
